flash/vision/classification/data.py

```diff
@@ -13,369 +13,392 @@
-        cls,
-        train_filepaths: Union[str, Optional[Sequence[Union[str, pathlib.Path]]]] = None,
-        train_labels: Optional[Sequence] = None,
-        train_transform: Optional[Callable] = _DEFAULT_TRAIN_TRANSFORMS,
-        valid_split: Union[None, float] = None,
-        valid_filepaths: Union[str, Optional[Sequence[Union[str, pathlib.Path]]]] = None,
-        valid_labels: Optional[Sequence] = None,
-        valid_transform: Optional[Callable] = _DEFAULT_VALID_TRANSFORMS,
-        test_filepaths: Union[str, Optional[Sequence[Union[str, pathlib.Path]]]] = None,
-        test_labels: Optional[Sequence] = None,
-        loader: Callable = _pil_loader,
-        batch_size: int = 64,
+    preprocess_cls = ImageClassificationPreprocess
+    image_size = (196, 196)
+
+    def __init__(
+        self,
+        train_ds: Optional[torch.utils.data.Dataset] = None,
+        valid_ds: Optional[torch.utils.data.Dataset] = None,
+        test_ds: Optional[torch.utils.data.Dataset] = None,
+        predict_ds: Optional[torch.utils.data.Dataset] = None,
+        batch_size: int = 1,
         num_workers: Optional[int] = None,
         seed: int = 1234,
+        train_split: Optional[Union[float, int]] = None,
+        valid_split: Optional[Union[float, int]] = None,
+        test_split: Optional[Union[float, int]] = None,
         **kwargs,
     ) -> 'ImageClassificationData':
-        """Creates a ImageClassificationData object from lists of image filepaths and labels
+        """Creates a ImageClassificationData object from lists of image filepaths and labels"""
 
-        Args:
-            train_filepaths: string or sequence of file paths for training dataset. Defaults to ``None``.
-            train_labels: sequence of labels for training dataset. Defaults to ``None``.
-            train_transform: transforms for training dataset. Defaults to ``None``.
-            valid_split: if not None, generates val split from train dataloader using this value.
-            valid_filepaths: string or sequence of file paths for validation dataset. Defaults to ``None``.
-            valid_labels: sequence of labels for validation dataset. Defaults to ``None``.
-            valid_transform: transforms for validation and testing dataset. Defaults to ``None``.
-            test_filepaths: string or sequence of file paths for test dataset. Defaults to ``None``.
-            test_labels: sequence of labels for test dataset. Defaults to ``None``.
-            loader: function to load an image file. Defaults to ``None``.
-            batch_size: the batchsize to use for parallel loading. Defaults to ``64``.
-            num_workers: The number of workers to use for parallelized loading.
-                Defaults to ``None`` which equals the number of available CPU threads.
-            seed: Used for the train/val splits when valid_split
+        if train_ds is not None and train_split is not None or valid_split is not None or test_split is not None:
```

Comment:
```suggestion
        if train_ds and train_split is not None or valid_split is not None or test_split is not None:
```
just curious can split be zero?

Reply:
Yes, it should be handled.